lib/delta.ex

```diff
@@ -141,9 +141,17 @@ defmodule Delta do
     middle
   end
 
-  def split(delta, 0), do: {[], delta}
+  def slice_max(delta, index, len) do
+    {_left, right} = split(delta, index, safe: true)
```

Comment:
The naming `:safe` might create incorrect associations - not 100% on the name but I'd suggest :align or :snap. Both of these kind of describe what it does (can change the exact position of index) but is also uncommon parameters so less likely to create an association